sw/applications/example_spi_host_dma/main.c

```diff
@@ -57,8 +57,16 @@ void dma_intr_handler_trans_done(void)
 }
 
 // Reserve memory array
-uint32_t flash_data[COPY_DATA_NUM] __attribute__ ((aligned (4))) = {0x76543210,0xfedcba98,0x579a6f90,0x657d5bee,0x758ee41f,0x01234567,0xfedbca98,0x89abcdef,0x679852fe,0xff8252bb,0x763b4521,0x6875adaa,0x09ac65bb,0x666ba334,0x44556677,0x0000ba98};
-uint32_t copy_data[COPY_DATA_NUM] __attribute__ ((aligned (4)))  = { 0 };
+#if SPI_DATA_TYPE == DMA_DATA_TYPE_DATA_TYPE_VALUE_DMA_32BIT_WORD
+    uint32_t flash_data[COPY_DATA_NUM] __attribute__ ((aligned (4))) = {0x76543210,0xfedcba98,0x579a6f90,0x657d5bee,0x758ee41f,0x01234567,0xfedbca98,0x89abcdef,0x679852fe,0xff8252bb,0x763b4521,0x6875adaa,0x09ac65bb,0x666ba334,0x44556677,0x0000ba98};
+    uint32_t copy_data[COPY_DATA_NUM] __attribute__ ((aligned (4)))  = { 0 };
+#elif SPI_DATA_TYPE == DMA_DATA_TYPE_DATA_TYPE_VALUE_DMA_16BIT_WORD
+    uint16_t flash_data[COPY_DATA_NUM] __attribute__ ((aligned (2))) = {0x7654,0xfedc,0x579a,0x657d,0x758e,0x0123,0xfedb,0x89ab,0x6798,0xff82,0x763b,0x6875,0x09ac,0x666b,0x4455,0x0000};
+    uint16_t copy_data[COPY_DATA_NUM] __attribute__ ((aligned (2)))  = { 0 };
+#else
+    uint8_t flash_data[COPY_DATA_NUM] = {0x76,0xfe,0x57,0x65,0x75,0x01,0xfe,0x89,0x67,0xff,0x76,0x68,0x09,0x66,0x44,0x00};
+    uint8_t copy_data[COPY_DATA_NUM] = { 0 };
+#endif
```

Comment:
Are these distinctions necessary? The same values can be used, we should just change how we cast the pointers when reading/writing (i.e. the word `0x76543210` can be accessed as 4 independent bytes `0x76` `0x54` ...by just accessing them with a `uint8_t*` instead of a `uint32_t*`) .

Reply:
agree, let me try

Reply:
done @JuanSapriza